api/envoy/config/rbac/v2/rbac.proto

```diff
@@ -121,8 +123,13 @@ message Permission {
 
     // A header (or pseudo-header such as :path or :method) on the incoming HTTP request. Only
     // available for HTTP request.
+    // Note: the pseudo-header :path includes the query and fragment string. Use the `url_path`
+    // field if you want to match the URL path without the query and fragment string.
```

Comment:
Would it make sense to either outright ban `:path` (and possibly other pseudo-headers?) or at least produce a warning if it's used in the `HeaderMatcher`?

Reply:
@mattklein123 and I discussed this in the private repo. I didn't add the warning because it seems a valid and legit use cases to use the pseudo-headers directly if the user wants the extra information. And it's also widely used in many other places in Envoy.

Also some pseudo-headers (e.g. :method) doesn't have any replacement for now.

Reply:
SGTM.